src/Model/Resolver/WishlistItemsResolver.php

```diff
@@ -132,26 +197,119 @@ public function resolve(
             $info
         );
 
+        $customerId = $wishlist->getCustomerId();
+        $storeId = $wishlist->getStore()->getId();
+
         $data = [];
+
+        /** @var \Magento\Wishlist\Model\Item\Interceptor $wishlistItem */
         foreach ($wishlistItems as $wishlistItem) {
             $wishlistItemId = $wishlistItem->getId();
             $wishlistProductId = $itemProductIds[$wishlistItemId];
             $itemProduct = $wishlistProducts[$wishlistProductId];
+            $type = $itemProduct['type_id'];
+
+            $price = $this->getItemPrice($wishlistItem, $type);
+            $priceWithoutTax = $this->getPriceWithoutTax(
+                $price, $itemProduct['model']->getData('tax_class_id'), $storeId, $customerId
+            );
+
+            $buyRequestOption = $wishlistItem->getOptionByCode('info_buyRequest');
+            $options = $this->getItemOptions($wishlistItem, $type);
 
             $data[] = [
                 'id' => $wishlistItemId,
                 'qty' => $wishlistItem->getData('qty'),
                 'sku' => $this->getWishListItemSku($wishlistItem),
+                'price' => $price,
+                'price_without_tax' => $priceWithoutTax,
+                'buy_request' => $buyRequestOption->getValue() ?? '',
                 'description' => $wishlistItem->getDescription(),
                 'added_at' => $wishlistItem->getAddedAt(),
                 'model' => $wishlistItem,
-                'product' => $itemProduct
+                'product' => $itemProduct,
+                'options' => $options
             ];
         }
 
         return $data;
     }
 
+    /**
+     * @param $item
+     * @param $type
+     * @return array
+     */
+    protected function getItemOptions($item, $type) {
+        $options = [];
+        switch ($type) {
+            case BundleType::TYPE_CODE:
+                $options = $this->bundleOptions->getOptions($item);
+                break;
+            case DownloadableType::TYPE_DOWNLOADABLE:
+                $options = $this->downloadableOptions->getOptions($item);
+                break;
+            default:
+                return $this->productOptions->getOptions($item);
+        }
+
+        $output = [];
+        foreach ($options as $option) {
+            $value = is_array($option['value']) ? join(', ', $option['value']) : $option['value'];
+            $output[] = [
+                'label' => $option['label'],
+                'value' => strip_tags($value)
+            ];
+        }
+        return $output;
+    }
+
+    /**
+     * @param Item $item
+     * @param string $type
+     * @return float
+     */
+    protected function getItemPrice($item, $type) {
+        $controller = self::PRICE_CALCULATION_MAP[$type] ?? null;
+        if ($controller === null) {
+            return -1;
+        }
```

Comment:
I guess it's better to either throw an error here or return null, to omit confusion of the future developers